panoply/datasource.py

```diff
@@ -7,9 +10,12 @@ class DataSource(events.Emitter):
     options = None
 
     # data source constructor
-    def __init__(self, source, options = {}):
-        super(DataSource, self).__init__()
-        
+    def __init__(self, source, options = {}, events = {}):
+        super(DataSource, self).__init__(events)
+
+        if 'destination' not in source:
```

Comment:
I don't see how things will break with the code I have added since it's the same code as we use elsewhere and it happens before the stream constructor (on it's parent constructor). Hence, every change of the `destination` in the stream constructor will be accepted.
Adding a method for changing the destination is an option, but it's a different discussion as the suggested implementation did not change the current one (developers should change the `source.destination` by themselves)